File: Session5/141. Linked List Cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        ListNode fptr=head;
        ListNode sptr=head;
        
        while(fptr!=null && fptr.next!=null)
        {
            sptr=sptr.next;
            fptr=fptr.next.next;
            if(sptr==fptr)
            {
                return true;
            }
        }
        return false;
    }
}

File: Session5/141. Linked List Cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        fstptr=head
        slptr=head
        
        if head is None:
            return False
        
        while(fstptr is not None and fstptr.next is not None):
            slptr=slptr.next
            fstptr=fstptr.next.next
            if slptr==fstptr:
                return True
        return False